### PR TITLE
Feature/extra cover tests

### PR DESCRIPTION
**Pull request by danoctavian**

### CODE REVIEW

File: contracts/modules/cover/Cover.sol [hunk 1]
```diff
@@ -416,7 +444,7 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     );
   }
 
-  function performPayoutBurn(
+  function performStakeBurn(
```
Comment (horodnicdragos) [inline contracts/modules/cover/Cover.sol:447]: What's the rationale behind this rename? Isn't "burnAmount" supposed to be the amount paid out (i.e. ETH/DAI, not NXM)? It definitely isn't the stake (in NXM) that is burned and we don't necessarily burn NXM when we make a payout, that's just a consequence of the payout.

File: contracts/modules/cover/Cover.sol [hunk 2]
```diff
@@ -301,7 +316,7 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     uint coverId,
     BuyCoverParams memory buyCoverParams,
     PoolAllocationRequest[] memory poolAllocations
-  ) external payable onlyMember whenNotPaused {
+  ) external payable onlyMember whenNotPaused nonReentrant {
```
Comment (roxdanila) [inline contracts/modules/cover/Cover.sol:319]: remove the require below:

`require(lastCoverSegment.start + lastCoverSegment.period > block.timestamp, "Cover: cover expired");`

Comment (danoctavian) [reply contracts/modules/cover/Cover.sol:321]: removed.

Changed condition in the IF to check for that.

File: contracts/modules/cover/Cover.sol (hunk 1 above)
Comment (danoctavian) [reply contracts/modules/cover/Cover.sol:452]: Discussed off-band.

**Review by roxdanila — APPROVED** [review]

**Review by roxdanila — APPROVED** [review]

**Pull request merged**